### PR TITLE
Code example improvements in Windows documentation

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR includes:
- Using explicit yaml+jinja code-blocks
- Capital letter to start a comment or description
- Use yes/no instead of true/false

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
Windows docs

##### ANSIBLE VERSION
v2.7

### CODE REVIEW

**Comment by dagwieers** [comment]
Can we disable those warnings ? It's not like those blocks aren't powershell.
These are false positives.

**Comment by dagwieers** [comment]
Looking at all the examples that hit this, all of them have backslashes in them.
It somehow seems that blackslashes are causing this ?

PS Also, the line-numbers seem to be from different files, and since **index.rst** seems to be a dummy name, it is very hard for contributors to understand what is going on. @felixfontein @gundalow

**Comment by dagwieers** [comment]
Pygments does not have any issues with the code-blocks, so it only fails through Sphinx.

**Comment by dagwieers** [comment]
Ok, it seems Sphinx run Pygments using `-F raiseonerror` as explained in sphinx-doc/sphinx#4225

And not the backslash itself seems to be the issue, but the colon is: http://pygments.org/demo/6759768/

**Comment by dagwieers** [comment]
And there is a fix to Sphinx here: sphinx-doc/sphinx#4250

**Comment by dagwieers** [comment]
Wrt. the powershell lexer, it has 2 issues with colons.

- It can't handle URLs that are not quoted, like:

      Set-Item -Path WSMan:\localhost\Shell\{path} -Value "value here"

  The solution here is simple, quote it as a string:

      Set-Item -Path "WSMan:\localhost\Shell\{path}" -Value "value here"

- It can't handle option/value using a colon as the delimiter

      winrs -r:http://server:5985/wsman -u:Username -p:Password ipconfig

  The solution is to write it as:

      winrs -r "http://server:5985/wsman" -u Username -p Password ipconfig

**Comment by dagwieers** [comment]
Our only true solution is to fix the lexer, but for the time being I have disabled Powershell syntax highlighting for the affected blocks. (I used "guess" which simply fails to use a lexer). I prefer doing this over working around it, so it is documented until fixed properly.

When we have a fix for the lexer, we can override the powershell lexer in our tree as we already do for the yaml+jinja lexer.

File: docs/docsite/rst/user_guide/windows_dsc.rst
Comment (jborean93) [inline docs/docsite/rst/user_guide/windows_dsc.rst:93]: I'm just wondering, is there a consensus as to whether we want upper case or lower case in the docs, I treat the name block as a bullet list so don't worry about starting with caps or ending with a sentence but that's just me.

File: docs/docsite/rst/user_guide/windows_dsc.rst
Comment (jborean93) [inline docs/docsite/rst/user_guide/windows_dsc.rst:239]: Same thing here, is it really necessary we do this?

**Review by jborean93 — COMMENTED** [review]
I'm not against the changes but I will leave it to @acozine as to whether we want to enforce a standard for comments/examples in the docs.

**Comment by jborean93** [comment]
@dagwieers you won't be able to do `winrs -r "http://server:5985/wsman" -u Username -p Password ipconfig` as `winrs` is a vbscript and require the args to be in the form currently documented. Happy for the `Set-Item -Path WSMan:\...` examples to be quoted though.

File: docs/docsite/rst/user_guide/windows_dsc.rst
Comment (dagwieers) [reply docs/docsite/rst/user_guide/windows_dsc.rst:93]: Hmm, I find it easier to read though. Especially for descriptions and documentation.

I don't mind that much for integration tests, but anything that ends up in the documentation (user-facing) I prefer to use sensible, pleasing standards.

File: docs/docsite/rst/user_guide/windows_dsc.rst
Comment (dagwieers) [reply docs/docsite/rst/user_guide/windows_dsc.rst:239]: Is it necessary ? From a technical perspective no, but to me it's more pleasing.

But I am the guy who prefers punctuation marks and correct spelling too ;-)

**Comment by dagwieers** [comment]
@jborean93 I am not changing any of them, I hope we can improve the lexer at some point. (However pygments seems to be unmaintained at the moment :-/)

File: docs/docsite/rst/user_guide/windows_dsc.rst
Comment (jborean93) [reply docs/docsite/rst/user_guide/windows_dsc.rst:93]: The trouble with using this as justification is that everybody has different preferences and we could get into an endless loop with these types of edits. If @acozine has a standard that she's trying to set with the documentation then we should be following that standard but right now I'm unsure what that would be.

Comment (samccann) [reply docs/docsite/rst/user_guide/windows_dsc.rst:93]: In general, Red Hat documentation follows the IBM Style guide. That said, it doesn't cover this scenario directly. Our Ansible style guide also doesn't cover this, but documenting modules has an example section:
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#examples-block

That uses an initial cap. Also the few examples I poked at in https://github.com/ansible/ansible-examples also uses initial cap for names.  I'll ask @acozine directly if she wants to adopt this as a standard (and say so in the docs so it's clear).

Comment (gundalow) [reply docs/docsite/rst/user_guide/windows_dsc.rst:93]: From a Community point of view we need to be careful not to cause "busy work" for Contributors and reviewers.

I think it's is OK to:

* Update "how to document your module" to state how we'd *like* documentation to be written
* update any examples in `docs/docsite/rst` to use the "best" form (with `name:`, capital letters, etc.)
* Bulk update of `support:core` modules to use the "best" form. 
* If others want to update modules that's fine, though that must be the only change in the PR to make it easy to review and backport. 

I wouldn't want to add review comments on PRs asking them to add capital letters (or fullstops/periods at the end).

Comment (samccann) [reply docs/docsite/rst/user_guide/windows_dsc.rst:93]: Chatted offline w/ @acozine and she doesn't have a preference for either way. So I think the 'gist' of it is:

- Don't comment on another PR to request capitalization etc change
- Recommend to always use a name and make it a useful phrase or sentence.
- no bulk changes to existing content, because there's no 'right or wrong' way to do it.

**Review by samccann — APPROVED** [review]
Generated the docs locally and looks good.

**Comment by samccann** [comment]
Docs looks okay. @jborean93 @gundalow anything else blocking the merge here?

**Comment by jborean93** [comment]
I'm ok with the technical content changes so +1 from me.

**Pull request merged**
